### ISSUE TITLE
Data Structures and Algorithms

### ISSUE THREAD

**Issue by abhikumar45444** [stackup-dev/open-source-campaign#227]
# Name of Proposer
*Dinesh*

# Campaign Name
*DSA - Effieciency of code matters*

## Campaign Description
*DSA aka Data structure and Algorithm, for a developer this will come in handy in every phase of your career. so what actually DSA is?  
A data structure is a storage that is used to store and organize data. It is a way of arranging data on a computer so that it can be accessed and updated efficiently.
- DSA will help you understand how to write th efficient code and also helps in increasing problem solving skill.
- Beginners or advance DSA will make you better developer.
- Data is very important in terms of everything so handling it properly is a duty of developer DSA teaches you that skill.*

## Number of Quests
*5*

## Quest 1 Summary
Introduction to DSA
- What is DS and why we should learn it.
- We will learn it through conepts and code chunks.*

## Quest 1 Learning Objectives
*By the end of this Quest, you will be familiar with Fundamentals of DSA.*

## Quest 1 Deliverables
*quiz*

## Quest 2 Summary
Array Datastructure
- What are arrays and how computer sees it.
- array as ADT and advantages.
- We will learn it through conepts and code visulaization.*

## Quest 2 Learning Objectives
*By the end of this Quest, you will be familiar with arrays and understand underlying concepts.*

## Quest 2 Deliverables
*quiz*

## Quest 3 Summary
Introduction to Linkedlist
- What is Linkedlist and how it is useful.
- Types of linkedlist and advantages.
- We will learn it through conepts and code visulaization.*

## Quest 3 Learning Objectives
*By the end of this Quest, you will be familiar with linkedlists.*

## Quest 3 Deliverables
*code screenshot*

## Quest 4 Summary
Introduction to Linear DS - Data structure in which data elements are arranged sequentially or linearly
- Types of Linear DS and Advantage.
- We will learn it through conepts and code chunks.*

## Quest 4 Learning Objectives
*By the end of this Quest, you will be familiar with Linear DS.*

## Quest 4 Deliverables
*code SS*

## Quest 5 Summary
Introduction to Non-Linear DS -Data structures where data elements are not placed sequentially or linearly are called non-linear data structures.
- Types of Non-Linear DS and Advantage.
- We will learn it through conepts and code chunks.*

## Quest 5 Learning Objectives
*By the end of this Quest, you will be familiar with Non-Linear DS.*

## Quest 5 Deliverables
*quiz*

**Comment by Amaankhn786** [comment]
Yes I also agree with your topic "Data Structures and Algorithms" is the topic which every coder needs in their carrier , it helps to write the effective code and increase the problem solving skills.
we have to cover mainly topic which are used in competitive programming and used in large company for interview.
submission should be in multiple languages like c++ , java , Python ..

**Comment by thomasgeorgec** [comment]
I think it is better to add a set of basic algorithms as well to the quests' list. learning both together will help in understanding more easily, as both are interrelated. it may include the very basic algorithms for searching, sorting to algorithms specific to individual Data structures, such as traversal, path finding.
